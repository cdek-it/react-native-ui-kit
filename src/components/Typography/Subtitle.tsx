import { memo } from 'react'
import { Text, type TextProps } from 'react-native'

import { makeStyles } from '../../utils/makeStyles'

export interface SubtitleProps extends TextProps {
  /**
   * true, если необходим базовый размер текста подзаголовка
   * @default false
   */
  base?: boolean
  /**
   * Выбор цвета подзаголовка
   * @default 'default'
   */
  color?: 'default' | 'primary' | 'secondary'
}

/**
 * Используется для подзаголовков
 * @see https://www.figma.com/design/2ZnL6XPKEpxAHvrlbRvnMu/Template-Tailwind-CSS-(DS)?node-id=1-245
 */
export const Subtitle = memo<SubtitleProps>(
  ({ base = false, color = 'default', style, ...other }) => {
    const styles = useStyles()

    return (
      <Text
        style={[styles.text, styles[color], base && styles.base, style]}
        {...other}
      />
    )
  }
)

const useStyles = makeStyles(({ theme }) => ({
  text: { fontSize: 12.25, fontWeight: 700, textTransform: 'uppercase' },
  base: { fontSize: 14 },
  default: { color: theme.General.textColor },
  primary: { color: theme.General.primaryColor },
  secondary: { color: theme.General.textSecondaryColor },
}))
