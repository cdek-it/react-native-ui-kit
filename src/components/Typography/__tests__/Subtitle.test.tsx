import { render } from '@testing-library/react-native'

import { Subtitle, type SubtitleProps } from '../Subtitle'

describe('Subtitle component tests', () => {
  const snapshotCases = generatePropsCombinations<Partial<SubtitleProps>>({
    base: [true, false, undefined],
    color: ['default', 'primary', 'secondary', undefined],
    style: [{ margin: 10 }],
  })

  test.each(snapshotCases)('base = $base, color = $color', (props) => {
    const renderedTitle = render(<Subtitle {...props}>Text</Subtitle>)

    expect(renderedTitle.toJSON()).toMatchSnapshot()
  })
})
