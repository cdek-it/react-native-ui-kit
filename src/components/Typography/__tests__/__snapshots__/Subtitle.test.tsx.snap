// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle component tests base = false, color = default 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = false, color = primary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "#188700",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = false, color = secondary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.6000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = false, color = undefined 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = true, color = default 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      {
        "fontSize": 14,
      },
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = true, color = primary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "#188700",
      },
      {
        "fontSize": 14,
      },
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = true, color = secondary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.6000)",
      },
      {
        "fontSize": 14,
      },
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = true, color = undefined 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      {
        "fontSize": 14,
      },
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = undefined, color = default 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = undefined, color = primary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "#188700",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = undefined, color = secondary 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.6000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;

exports[`Subtitle component tests base = undefined, color = undefined 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "fontWeight": 700,
        "textTransform": "uppercase",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      false,
      {
        "margin": 10,
      },
    ]
  }
>
  Text
</Text>
`;
